admin/adminChatAnalyser.php

```diff
@@ -0,0 +1,201 @@
+	{
+		if (in_array($Game->Members->ByUserID[$User->id]->countryID,$countryID1) && !(in_array($Game->Members->ByUserID[$User->id]->countryID,$countryID2)))
+			$countryID1=array($Game->Members->ByUserID[$User->id]->countryID);
+		elseif (in_array($Game->Members->ByUserID[$User->id]->countryID,$countryID2) && !(in_array($Game->Members->ByUserID[$User->id]->countryID,$countryID1)))
+			$countryID2=array($Game->Members->ByUserID[$User->id]->countryID);
+		elseif (!(in_array($Game->Members->ByUserID[$User->id]->countryID,$countryID2)) && !(in_array($Game->Members->ByUserID[$User->id]->countryID,$countryID1)))
+			$countryID1=$countryID2=array();
+		else
+			$countryID1=$countryID2=array($Game->Members->ByUserID[$User->id]->countryID);
+	}
+	
+	print $Game->titleBar().'<br>';
+	
+	print '<DIV class="variant'.$Variant->name.'">
+		<TABLE>
+			<THEAD>
+				<TH style="border: 1px solid #000">Country</TH>
+				<TH style="border: 1px solid #000">Player</TH>
+				<TH style="border: 1px solid #000" align="center">Country 1</TH>
+				<TH style="border: 1px solid #000" align="center">Country 2</TH>
+			</THEAD>
+			<TR>
+				<TD style="border: 1px solid #666">Global</TD>
+				<TD style="border: 1px solid #666"></TD>
+				<TD style="border: 1px solid #666" align="center"><input type="checkbox" name="countryID1[]" value="0" '.(in_array('0',$countryID1)?'checked':'').'></TD>
+				<TD style="border: 1px solid #666" align="center"><input type="checkbox" name="countryID2[]" value="0" '.(in_array('0',$countryID2)?'checked':'').'></TD>
+			</TR>';
+
+	foreach($Game->Members->ByCountryID as $Member)
+		print '
+			<TR>
+				<TD style="border: 1px solid #666"><span class="memberCountryName">'.$Member->memberCountryName().'</TD>
+				<TD style="border: 1px solid #666">'.$Member->memberUserDetail().'</TD>
+				<TD style="border: 1px solid #666" align="center"><input type="checkbox" name="countryID1[]" value="'.$Member->countryID.'" '.(in_array($Member->countryID,$countryID1)?'checked':'').'></TD>
+				<TD style="border: 1px solid #666" align="center"><input type="checkbox" name="countryID2[]" value="'.$Member->countryID.'" '.(in_array($Member->countryID,$countryID2)?'checked':'').'></TD>
+			</TR>';
+				
+	print '</TABLE></DIV>';
+}
+
+print '<br><input type="submit" name="Submit" class="form-submit" value="Check" /></form>';
+
+if (($gameID != 0) && (count($countryID1) > 0) && (count($countryID2) > 0))
+{
+	$sql = 'SELECT message, toCountryID, fromCountryID, turn, timeSent
+				FROM wD_GameMessages WHERE
+					gameID = '.$gameID.' AND
+					(
+						( toCountryID IN ('.implode(', ',$countryID1).') AND fromCountryID IN ('.implode(', ',$countryID2).') )
+							OR
+						( toCountryID IN ('.implode(', ',$countryID2).') AND fromCountryID IN ('.implode(', ',$countryID1).') )
+					)
+				ORDER BY id ASC';
+	
+	$tabl = $DB->sql_tabl($sql);
+	
+	print '<BR><DIV class="variant'.$Variant->name.'">';
+	print '<TABLE class="chatbox">';
+	
+	$alternate = $output = false;
```

Comment:
This seems misleading; I think the variables should be declared separately, because they are unrelated.